src/node_shapes.jl

```diff
@@ -0,0 +1,227 @@
+    node_text(text::LaTeXString, offset::Point, color::String)
+
+Label a node with text, which is also rotated to match the direction specified by `offset`.
+
+# Arguments
+- `offset::Point`: Specify the offset of the text label with respect to the node center.
+    - The text is translated and rotated in the direction of the offset.
+- `color`: Used only when `align` is not `:inside`. Specifies an offset in the direction of `align`.
+"""
+function node_text(text::AbstractString, offset::Point, color::String="black")
+    return (args... ; text=text, position=O, text_color=color, clip=false, kwargs...) -> begin
+        clip ? clipreset() : nothing
+        sethue(text_color)
+        translate(position+offset)
+        rotate(slope(O, offset)+pi/2)
+        Luxor.text(text, valign=:middle, halign=:center)
+        rotate(-slope(O, offset)-pi/2)
+        translate(-position-offset)
+    end
+end
+
+function node_text(text::LaTeXString, offset::Point, color::String="black")
+    return (args...; text=text, position=O, text_color=color, clip=false, kwargs...) -> begin
+        clip ? clipreset() : nothing
+        sethue(text_color)
+        translate(position+offset)
+        rotate(slope(O, offset)+pi/2)
+        latex(text, O, :middle, :center)
+        rotate(-slope(O, offset)-pi/2)
+        translate(-position-offset)
+    end
+end
+
+"""
+    node_fill(fill::Symbol, arg::String)
+
+Fill in the interior of the node with an image or a color.
+
+# Arguments
+- `fill::Symbol`: Can be `image` or `color`.
+- `arg::String`: Can be a color like "red" or a image path in the same directory like "image.png".
+"""
+function node_fill(fill::Symbol, arg::String)
+    opts = Dict{Symbol, Any}()
+    opts[:fill] = fill
+    if fill == :image
+        draw = (args...; image_path=arg, bounding_box=(O, O), clip=false, kwargs...) -> begin
+            img = readpng(image_path)
+            image_dims = last(bounding_box)-first(bounding_box)
+            scl = image_dims/(img.width, img.height)
+            scale(scl...)
+            placeimage(img, bounding_box[1]/Tuple(scl))
+            scale(1/scl[1], 1/scl[2])
+            clip ? clipreset() : nothing
+        end
+    elseif fill == :color
+        draw = (args...; fill_color=arg, bounding_box=(O, O), clip=false, kwargs...) -> begin
+            sethue(fill_color)
+            fill_dims = last(bounding_box)-first(bounding_box)
+            circle((bounding_box[1]+bounding_box[2])/2, max(fill_dims...)/2, :fill)
+            clip ? clipreset() : nothing
+        end
+    end
+    return opts, draw
+end
+
+"""
+    node_border(color::String, border_width::Real)
+
+Create a border around the node shape.
+
+If the `:shape` is not specified, the `bounding_box` becomes the border.
+
+# Arguments
+- `color::Symbol`
+- `border_width::Real`: The line width used for the border. Defualt is 3.
+"""
+function node_border(color::String, border_width::Real=3)
```

Comment:
There is a mismatch in the `color` type here. You sometimes write `Symbol` and sometimes `String`. I think we shouldn't have any type here as we want to allow all kinds of colors.